docker-compose.yml

```diff
@@ -0,0 +1,20 @@
+services:
+  web:
+    build:
+      context: .  # Build from the project root
+    ports:
+      - "8000:8000"
+    volumes:
+      - .:/app  # Mount the project root into /app
+    environment:
+      - PYTHONPATH=/app
+    command: uvicorn api.app:app --host 0.0.0.0 --port 8000
+
+  tests:
```

Comment:
Running these tests should not happen here. This should be in a GitHub actions workflow.yml file.

1. In GitHub Actions, define a workflow that runs pytest inside a container.
2. In the production environment, after tests succeed, build the Docker image and deploy it. Build the docker image only if the tests pass.